save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/OrganizationController.kt

```diff
@@ -362,8 +367,17 @@ internal class OrganizationController(
         .switchIfEmptyToResponseException(HttpStatus.FORBIDDEN) {
             "Not enough permission for managing organization git credentials."
         }
-        .map {
-            gitService.delete(it, url)
+        .map { organization ->
+            // Find and remove all corresponding data to the current git repository
+            val git = gitService.getByOrganizationAndUrl(organization, url)
+            val testSuitesSources = testSuitesSourceService.findByGit(git)
+            testSuitesSources.forEach { testSuitesSource ->
+                val testSuites = testSuitesService.getBySource(testSuitesSource)
+                testSuitesService.deleteTestSuiteDto(testSuites.map { it.toDto() })
+                testSuitesSourceService.delete(testSuitesSource)
```

Comment:
we also have `TestSuitesSourceStorage` which also should be cleaned after deleting `TestSuitesSource`